src/Microsoft.Extensions.Hosting.Abstractions/IHostLifetime.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Extensions.Hosting
+{
+    public interface IHostLifetime
+    {
+        /// <summary>
+        /// Called at the start of <see cref="IHost.StartAsync(CancellationToken)"/> to see if the host is ready to start.
+        /// This can be used to delay startup until signaled by an external event.
+        /// </summary>
+        /// <param name="cancellationToken">Used to indicate that StartAsync has been aborted.</param>
+        Task WaitForStartAsync(CancellationToken cancellationToken);
```

Comment:
After discussion with @Tratcher , we should revert this to the original interface definition.